HP-1526: New layout

NikoHelle opened the pull request:
New layout based on https://www.figma.com/proto/WaYKxyMHA5Gze1lmJICLbI/%F0%9F%91%A4-Helsinki-profiili?page-id=0%3A1&node-id=2%3A2&viewport=899%2C639%2C1&scaling=min-zoom

Biggest changes are:
- white boxes are gone
- download and delete profile boxes are not accordions anymore (can't open/close)
- bottom page under download and delete profile components is gray

Commit:
(HP-1526) Unbox DownloadData content

Removed the wrapping Accordion.

Commit:
(HP-1526) Enable deleteMyProfile in mocked graphQL queries

Commit:
(HP-1526) Handle service connection query error in when delete starts

Commit:
(HP-1526) Unbox DeleteProfile content

Also removed confirmation checkbox from DeleteProfile

There is already a confirmation modal, no need for double check

Commit:
(HP-1526) No need for a grid wrapper element anymore

Commit:
(HP-1526) Change bg colors and component alignment

White on white was removed. The content area gets a litte wider as box padding is gone.

ProfileInformation bottom area needs to be grey all the way to the sides.

Removed user name with koros

To fill full width of the page, the content padding must be moved inside components. And components must be wrapped.

So ViewProfile has no css or extra elements anymore and those are moved to ServiceConnections and ProfileInformation.

Fixed tests. User name in "view-profile-heading" was removed and caused failures. Added new testIds.

Commit:
(HP-1526) Change main bg color to white

Commit:
(HP-1526) Restyled AuthenticationProviderInformation

Css became obsolete.

Commit:
(HP-1526) Moved email component down.

Commit:
(HP-1526) Change AccessibilityStatement styles to match other changes

No need for white bg, boxes or padding.

Setting width explicitly for desktop is obsolete after setting  max-width

Commit:
(HP-1526) Fixed mobile view padding and width problems

EmailEditor resized different way than other

Language selector had no max-width.

actions-wrapper bottom padding made element padding inconsistent.

Commit:
(HP-1526) Re-styled expanding panel

White bg on white did not emphasize the box so added light borders

Commit:
(HP-1526) Removed green checkbox icon from VerifiedPersonalInformation

Icon is never used, so removed it from LabeledValue too

Commit:
(HP-1526) Added hook for scrolling elements into view

Added scrollIntoView in to Jest set up

Jest did not have this function and tests would fail.

Commit:
(HP-1526) Added downloadMyProfile to mocked response data

Commit:
(HP-1526) Added gdprQueryScope and gdprDeleteScope to mocked data

DeleteProfile did not need these because it does not use cache.

DownloadProfile threw errors when cache was updated without these props.

Commit:
(HP-1526) Improve padding

Padding in mobile views were too tight or non-existing.

Unified paddings between ServiceConnection and ViewProfile pages.

akikoskinen commented:
The service connections are fetched from BE many times unnecessarily. Click "delete profile" (but don't actually delete the profile). Go to "Your services" tab, return to information tab. Go to "delete profile" again (without deleting it this time either). The service connections are fetched many times during these steps while one time would be enough.

NikoHelle commented:
I had to add `fetchPolicy: 'no-cache'` to line 77 in `DeleteProfile.tsx`. This causes reload every time user navigates to the Profile -page and clicks the delete button. (Which is not very often).

Without "no-cache", the Apollo-client does not trigger `onComplete`-callback, if data is already in cache.

Same has been done before in this commit:
https://github.com/City-of-Helsinki/open-city-profile-ui/commit/4f1d9de6dba7aaef51890e2d58b6459da36b1cdc

Previously caching did not matter, because service connections were loaded before user can press the delete button and confirm the deletion from the confirmation modal.

Now service connections are loaded after delete is pressed and confirmation modal is showed after that. When cached, the modal would never been show as `onComplete`-callback would not been called.

Very annoying. At least service connections are not fetched every time delete is pressed. Only if user navigates away from that page and tries to delete again.

akikoskinen commented:
Click "delete profile", but cancel the modal. Change language. Profile deletion modal pops up again.

NikoHelle commented:
The "onComplete"-callback is called again when component is re-rendered after language change. Happens when serviceConnections have already been fetched.

Added a check for that and it is not happening anymore.

akikoskinen commented on src/profile/components/deleteProfile/DeleteProfile.tsx:
```diff
@@ -67,13 +67,21 @@ function DeleteProfile(): React.ReactElement {
+        return;
+      }
       setDataLoadState(loadedLoadState);
+      // eslint-disable-next-line @typescript-eslint/no-use-before-define
```
Is the `@typescript-eslint/no-use-before-define` rule something that should be followed (i.e. reorganize definitions above usages), or can the rule just be disabled in eslint configuration?

NikoHelle replied:
Sometimes functions cannot be ordered by usage, because they cross-use each other and one usage is always before the definition. 

In this case re-ordering can be done

akikoskinen commented on src/auth/components/login/Login.module.css:
```diff
@@ -4,6 +4,9 @@
   flex: 1;
+  background-color: var(--color-black-ultra-light);
+  /* footer koros is 14px high and "bleeds" bg color. This shifts the koros up and hides the bleed */
+  margin-bottom: -14px;
```
Login page has this negative margin to put the background color "between the waves" of the footer koros. But the profile information page doesn't have this. There the "between the waves" is white, while the bottommost parts (delete profile) of the main content background color is light gray.

NikoHelle replied:
Fixed! Moved the margin to base element of all content areas

akikoskinen commented on src/profile/components/deleteProfile/DeleteProfile.tsx:
```diff
@@ -149,48 +154,37 @@ function DeleteProfile(): React.ReactElement {
-        )}
-      </ExpandingPanel>
-
+    <ProfileSection hasVerifiedUserData data-test-id={'delete-profile'}>
```
I suppose the `hasVerifiedUserData` value has lost its original meaning. Now it seems to be used to mean "use `background: none` CSS style". The value and perhaps also the `.verified-data` class should probably be renamed.

or

If `hasVerifiedUserData` is `false`, then `ProfileSection` gets a white background. This seems to be used in places where the underlying background color is also white, so it doesn't add anything visible. Can the white background and `hasVerifiedUserData` flag just be removed from `ProfileSection`?

NikoHelle replied:
True, good catch. Removed the css and usages.

akikoskinen replied:
Shouldn't the white background now be removed from the `.profile-section` style, making `ProfileSection`s transparent?

NikoHelle replied:
Yes, and also useless border-radius is now also removed. 

Removed also TopSectionWithKoros component because it is not used anywhere.

Commit:
(HP-1526) Remove Koros from CreteProfile

Match layout with ViewProfile

Commit:
(HP-1526) Header menu items should not be shown when creating a profile

Commit:
(HP-1526) Removed usess css class verified-data

background none is useless now when whole page bg is white.

Removed all usages via "hasVerifiedUserData" prop

Commit:
(HP-1526) Version bump

Commit:
(HP-1526) Removed unused TopSectionWithKoros
